src/jira.ts

```diff
@@ -0,0 +1,37 @@
+import {Version3Client} from 'jira.js'
+import * as core from '@actions/core'
+export interface JiraConfig {
+  host: string
+  email: string
+  apiToken: string
+}
+
+export interface JiraClient {
+  issueExists: (issueIdOrKey: string) => Promise<boolean>
+}
+
+export class JiraClientImpl implements JiraClient {
+  private readonly client: Version3Client
+
+  constructor({host, email, apiToken}: JiraConfig) {
+    this.client = new Version3Client({
+      host,
+      authentication: {
+        basic: {
+          email,
+          apiToken
+        }
+      }
+    })
+  }
+
+  async issueExists(issueIdOrKey: string): Promise<boolean> {
+    try {
+      await this.client.issues.getIssue({issueIdOrKey})
```

Comment:
Does this err if there's no issue with that id?

Reply:
it 404s and throws if the issue does not exist

Reply:
I suppose if it 401 auth error the issue could still exist, but not sure if that matters here

Reply:
Wow....horrible API design causing exception handling as flow control :smdh: